cpp/src/groupby/common/aggregation_requests.hpp

```diff
@@ -52,6 +53,33 @@ using AggRequestType = std::pair<gdf_column*, operators>;
  *---------------------------------------------------------------------------**/
 using SimpleAggRequestCounter = std::pair<AggRequestType, gdf_size_type>;
 
+
+static constexpr std::array<operators, 4> simple_aggregations = {SUM, MIN, MAX, COUNT};
+
+static constexpr std::array<operators, 2> complex_aggregations = {MEDIAN, QUANTILE};
+
+
+/**---------------------------------------------------------------------------*
+ * @brief  To verify that the input operator is part of simple_aggregations list.
+ * Note that in this kind of aggregators can be computed in a single pass scan.
+ * In the other hand compound aggregation like AVG need to be computed by simple
+ * ones (SUM and COUNT).
+ *---------------------------------------------------------------------------**/
+inline bool is_simple(operators op) {
+  return std::any_of(simple_aggregations.begin(), simple_aggregations.end(),
+                     [&](operators test) { return test == op; });
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief  To verify that the input operator is part of  complex_aggregations list.
+ * Complex aggregation is used to identify other ones like MEDIUM and  QUANTILE,
```

Comment:
Not sure about the name "Complex" -- it's too similar to "Compound", and could be confused with complex numbers. I think the type of aggregations we're considering here require ordering the inputs. I think they should be called "ordered aggregations".

There's another type that is unordered, but produces O(N) output rather than O(1) output. An example is `cumsum` (prefix sum). Another is `unique`. That is a different type again, so I don't think everything can be grouped as "Complex".